lib/travis/api/v3/models/repository.rb

```diff
@@ -98,5 +98,10 @@ def debug_tools_enabled?
       return true if Travis::Features.active?(:debug_tools, self)
       return false
     end
+
+    def managed_by_installation
+      return true if read_attribute(:managed_by_installation_on)
+      return false
+    end
```

Comment:
OK, so the timestamp field on a repository is used to differentiate repos that are managed by installation rather than by service hook?

How is this being set?
Do we expect repositories to not be using installations for an organisation that has an installation?

This method could also be shortened to:

``` ruby
def managed_by_installation
  !!managed_by_installation_on
end
```

Timestamp fields are traditionally called `…_at`, not `…_on`, though maybe we haven't been sticking to that.

However, I would suggest to not support repositories via service hooks if the owner has a GitHub installation. If that is the case, then I think we don't need a field in the database.

Instead I would approach it something like this:

``` ruby
class Models::Installation < Model
  belongs_to :owner, polymorphic: true
end

class Models::User < Model
  has_one :installation, as: :owner
end

class Models::Organization < Model
  has_one :installation, as: :owner
end

class Models::Repository < Model
  has_one :installation, through: :owner

  def managed_by_installation?
    installation.present?
  end
end
```